hail_search/queries/base.py

```diff
@@ -321,60 +330,50 @@ def import_filtered_table(self, project_samples, num_families, intervals=None, *
-                        comp_het_families_ht, comp_het_project_ht,
-                        default=hl.range(num_families).map(lambda i: hl.missing(entry_type)),
-                        default_1=hl.range(num_project_families).map(lambda i: hl.missing(entry_type)),
-                    )
-                if ht_added:
-                    num_families += num_project_families
-                    num_projects_added += 1
-
-            if len(filtered_project_hts) > num_projects_added:
-                logger.info(f'Found {num_projects_added} {self.DATA_TYPE} projects with matched entries')
+            families_ht, comp_het_families_ht = self._load_filtered_project_hts(project_samples, **kwargs)
 
         if comp_het_families_ht is not None:
             self._comp_het_ht = self._query_table_annotations(comp_het_families_ht, self._get_table_path('annotations.ht'))
-            if not use_annotations_ht_first:
-                self._comp_het_ht = self._filter_annotated_table(self._comp_het_ht, is_comp_het=True, **kwargs)
+            self._comp_het_ht = self._filter_annotated_table(self._comp_het_ht, is_comp_het=True, **kwargs)
             self._comp_het_ht = self._filter_compound_hets()
 
         if families_ht is not None:
             self._ht = self._query_table_annotations(families_ht, self._get_table_path('annotations.ht'))
-            if not use_annotations_ht_first:
-                self._ht = self._filter_annotated_table(self._ht, **kwargs)
-            elif self._has_comp_het_search:
-                self._ht = self._filter_by_annotations(self._ht, **(kwargs.get('parsed_annotations') or {}))
+            self._ht = self._filter_annotated_table(self._ht, **kwargs)
+
+    def _filter_merged_project_hts(self, project_hts, sample_data, filtered_project_hts, filtered_comp_het_project_hts, n_partitions, **kwargs):
+        if not project_hts:
+            return
+        ht = self._merge_project_hts(project_hts, n_partitions, include_all_globals=True)
+        ht, comp_het_ht = self._filter_entries_table(ht, sample_data, **kwargs)
+        if ht is not None:
+            filtered_project_hts.append(ht)
+        if comp_het_ht is not None:
+            filtered_comp_het_project_hts.append(comp_het_ht)
 
-    def _add_project_ht(self, families_ht, project_ht, default, default_1=None):
-        if default_1 is None:
-            default_1 = default
-
-        if not project_ht.any(project_ht.family_entries.any(hl.is_defined)):
-            return families_ht, False
-
-        families_ht = families_ht.join(project_ht, how='outer')
-        families_ht = families_ht.select_globals(
-            family_guids=families_ht.family_guids.extend(families_ht.family_guids_1)
+    @staticmethod
+    def _merge_project_hts(project_hts, n_partitions, include_all_globals=False):
+        if not project_hts:
+            return None
+        ht = hl.Table.multi_way_zip_join(project_hts, 'project_entries', 'project_globals')
```

Comment:
I'm not positive on this, but you might want to ensure that all tables in `project_hts` have the same # of partitions for `multi_way_zip_join` to work the best it can - you could pass some constant value as `n_partitions` to `_read_table`

Reply:
All the tables are read in with a specified interval, and there is no way to specify the interval and the number of partitions, hail throws an error. But they generally are all 1 partition because the interval is pretty small, even for the largest of tables